circus-train-s3-s3-copier/src/main/java/com/hotels/bdp/circustrain/s3s3copier/aws/TransferManagerFactory.java

```diff
@@ -26,12 +31,19 @@
 @Component
 public class TransferManagerFactory {
 
+  private static final Logger LOG = LoggerFactory.getLogger(TransferManagerFactory.class);
+
   public TransferManager newInstance(AmazonS3 targetS3Client, S3S3CopierOptions s3s3CopierOptions) {
-    return TransferManagerBuilder
-        .standard()
+    TransferManagerBuilder builder = TransferManagerBuilder.standard()
         .withMultipartCopyThreshold(s3s3CopierOptions.getMultipartCopyThreshold())
         .withMultipartCopyPartSize(s3s3CopierOptions.getMultipartCopyPartSize())
-        .withS3Client(targetS3Client)
-        .build();
+        .withS3Client(targetS3Client);
+
+    if (s3s3CopierOptions.getMaxThreadPoolSize() != -1) {
```

Comment:
Can we add a unit test for this if statement or is it not straightforward to check the existence of the executor factory?

Reply:
I had originally added the transfermanagerFactoryTest to cover it but found that it wasn't straightforward. That property is directly set on the executor factory which transferManager stores under a private final w/o any getters.